src/allencell_ml_segmenter/main/viewer.py

```diff
@@ -105,3 +129,79 @@ def _get_layer_by_name(self, name: str) -> Optional[Layer]:
             if l.name == name:
                 return l
         return None
+
+    def get_seg_layers(self) -> list[Layer]:
+        """
+        Get only segmentation layers (which should be probability mappings) from the viewer.
+        These are the layers that start with [seg].
+        """
+        return [
+            layer
+            for layer in self.get_layers()
+            if layer.name.startswith("[seg]")
+        ]
+
+    def insert_binary_map_into_layer(
+        self,
+        layer: Layer,
+        image: np.ndarray,
+        remove_seg_layers: bool = False,
+    ) -> None:
+        """
+        Insert a binary mpa image into the viewer.
+        If a layer for this binary map image already exists, the new image will replace the old one and refresh the viewer.
+        If the layer does not exist, it will be added to the viewer in the correct place (on top of the original raw image:
+        index_of_segmentation + 1 in the LayerList)
+
+        :param layer: layer to replace.
+        :param image: image to insert
+        :param remove_seg_layers: boolean indicating if the layer that is being thresholded is a segmentation layer, and should be removed from the layer once it is updated with the threshold.
+        """
+        # if threshold has not been previously applied, update name
+        if (
+            "threshold_applied" not in layer.metadata
+            or not layer.metadata["threshold_applied"]
+        ):
+            layer.name = f"[threshold] {layer.name}"
+        layer.data = image
+        layer.metadata["threshold_applied"] = True
+        layer.refresh()
+
+    def get_source_path(self, layer: Layer) -> Optional[Path]:
+        """
+        Given a layer, gets that layer's source path- the path to the image on the filesystem.
+        """
+        # If the image was dragged into napari, we expect a layer.source.path
+        if layer.source.path is not None:
+            return Path(layer.source.path)
+        # if the image was added by the plugin, we expect source_path to be set instead
+        if layer.metadata is not None and "source_path" in layer.metadata:
+            return Path(layer.metadata["source_path"])
+        return None
+
+    def get_all_layers_containing_prob_map(self) -> list[Layer]:
+        """
+        Get all segmentation labels layers that currently exist in the viewer.
+        """
+        return [
+            layer
+            for layer in self.get_layers()
+            if "prob_map" in layer.metadata
+        ]
+
+    def clear_binary_map_from_layer(self, layer: Layer) -> None:
+        """
+        We need to keep the layer because it contains the segmentation's probability map.
+        So, clear out the binary map by setting it to all zeros
+        """
+        if "threshold_applied" in layer.metadata:
```

Comment:
These metadata keys are hardcoded in a few places I imagine? If so extracting constants would good.